common/util/file_util.h

```diff
@@ -34,6 +34,12 @@ absl::string_view Basename(absl::string_view filename);
 // there is no "." in the basename, this is equivalent to file::Basename(path).
 absl::string_view Stem(absl::string_view filename);
 
+// Search for 'filename' file starting at 'start' (can be file or directory)
+// and going upwards the directory tree. Returns true if anything is found and
+// puts the found file in 'result'.
+bool UpwardFileSearch(std::string start, std::string filename,
```

Comment:
This is an util so it can be used by others too (might be useful e.g. for waiver files if the decision will be to implement them).